app/web_framework/cluster_master_application.py

```diff
@@ -30,58 +31,100 @@ def __init__(self, cluster_master):
+            RouteNode(r'version', _VersionHandler2),
+            RouteNode(r'builds', _BuildsHandler).add_children([
+                RouteNode(r'(\d+)', _BuildHandler, 'build').add_children([
+                    RouteNode(r'result', _BuildResultRedirectHandler),
+                    RouteNode(r'artifacts.tar.gz', _BuildTarResultHandler),
+                    RouteNode(r'artifacts.zip', _BuildZipResultHandler),
+                    RouteNode(r'subjobs', _SubjobsHandler,).add_children([
+                        RouteNode(r'(\d+)', _SubjobHandler, 'subjob').add_children([
+                            RouteNode(r'atoms', _AtomsHandler).add_children([
+                                RouteNode(r'(\d+)', _AtomHandler, 'atom').add_children([
+                                    RouteNode(r'console', _AtomConsoleHandler)
+                                ])
+                            ]),
+                            RouteNode(r'result', _SubjobResultHandler)
+                        ])
+                    ])
+                ])
+            ]),
+            RouteNode(r'queue', _QueueHandler),
+            RouteNode(r'slaves', _SlavesHandler).add_children([
+                RouteNode(r'(\d+)', _SlaveHandler, 'slave').add_children([
+                    RouteNode(r'shutdown', _SlaveShutdownHandler)
+                ]),
+                RouteNode(r'shutdown', _SlavesShutdownHandler)
+            ]),
+            RouteNode(r'eventlog', _EventlogHandler)]
+
+        api_old_v1 = [
+            # Make sure to do a deep copy of other routes so parents don't get reassigned in original list.
+            RouteNode(r'v1', _APIVersionOneHandler).add_children([deepcopy(route) for route in api_v1])]
+
+        root = RouteNode(r'/', _RootHandler)
+
+        # Include routes starting with `v1` for backwards compatibility
+        root.add_children(api_old_v1, version=1)
+        root.add_children(api_v1, version=1)
+        root.add_children(api_v2, version=2)
+
+        handlers = self.get_all_handlers(root, default_params)
         super().__init__(handlers)
 
 
 class _ClusterMasterBaseAPIHandler(ClusterBaseAPIHandler):
-    def initialize(self, route_node=None, cluster_master=None):
+    def initialize(self, route_node_wrapper=None, cluster_master=None):
         """
-        :type route_node: RouteNode
+        :type route_node_wrapper: RouteNodeWrapper
         :type cluster_master: app.master.cluster_master.ClusterMaster
         """
         self._logger = log.get_logger(__name__)
         self._cluster_master = cluster_master
-        super().initialize(route_node)
+        super().initialize(route_node_wrapper)
 
 
 class _RootHandler(_ClusterMasterBaseAPIHandler):
-    pass
+    def get(self):
+        response = {
+            'master': self._cluster_master.api_representation(),
+        }
+        self.write(response)
 
 
-class _APIVersionOneHandler(_ClusterMasterBaseAPIHandler):
+class _APIVersionOneHandler(_RootHandler):
```

Comment:
I kind of assumed we wanted to do this -- basically the master info was only available on the `v1` endpoint so I moved it to the root endpoint and I have the `v1` endpoint mimic it for backwards compatibility